### PR TITLE
Bugfix/FOUR-9727: Non-system assets are being hidden from the UI as if they were system assets

### PR DESCRIPTION
**Pull request by agustinbusso**
## Issue & Reproduction Steps
Please see to the ticket https://processmaker.atlassian.net/browse/FOUR-9727

## Solution
- Update the scopeNonSystem function to look at category_assignments to make sure the asset does not belong to a system category.

**Working video**

https://github.com/ProcessMaker/processmaker/assets/90727999/07e6dd84-5e70-494a-b2ef-a0e8fa045927

## How to Test
Test 1
- Import the attached process in the ticket
- Go to scripts and verify that the script is showing in the list

Test 2
- In category_assignments table, modify the category_id to point to a system category (you need to go to script_categories table and modify some script category to be is_system = 1)
- Go back to scripts, and the script using the system category should not show

Test 3
- Test the same as test 2 but with processes
- Test the same as test 2 but with screens
- Test the same as test 2 but with data sources

## Related Tickets & Packages
- [FOUR-9727](https://processmaker.atlassian.net/browse/FOUR-9727)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

[FOUR-9727]: https://processmaker.atlassian.net/browse/FOUR-9727?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ProcessMaker/Traits/HideSystemResources.php [hunk 1]
```diff
@@ -59,37 +59,41 @@ public function scopeNonSystem($query)
...
-                    ->where('is_template', false)
-                    ->when(Schema::hasColumn('screens', 'asset_type'), function ($query) {
-                        return $query->whereNull('asset_type');
-                    });
+            return $query
+                ->where('is_template', false)
+                ->when(Schema::hasColumn('screens', 'asset_type'), function ($query) {
+                    return $query->whereNull('asset_type');
+                })
+                ->whereDoesntHave('categories', function ($query) {
```
Comment (eiresendez) [inline ProcessMaker/Traits/HideSystemResources.php:76]: hi @agustinbusso this code displays the system screens (which should be hidden) can you try this query instead?

```
->whereHas('categories', function ($query) {
    $query->where('is_system', false);
})
```

image

Comment (agustinbusso) [reply ProcessMaker/Traits/HideSystemResources.php:76]: Good catch, thank you so much Eleazar!

**Review by eiresendez — APPROVED** [review]
approved! Tested for scripts, screens, processes and data sources, both manually and with packages that installed resources by default.

**Review by eiresendez — CHANGES_REQUESTED** [review]

File: upgrades/2023_08_02_168972_add_default_screens_to_system_category.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class AddDefaultScreensToSystemCategory extends Upgrade
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        $this->assignToSystemCategory('default-display-screen');
+        $this->assignToSystemCategory('default-form-screen');
+    }
```
[the remark this answers is not in the document]
Comment (agustinbusso) [reply upgrades/2023_08_02_168972_add_default_screens_to_system_category.php:16]: @eiresendez yesterday we sync with @nolanpro, users should be able to modify interstitials. The only two screens they not supposed to edit are default display screen and default form screen

**Review by eiresendez — APPROVED** [review]

**Pull request merged**